### ISSUE TITLE
Building Tracy convenience target requires LLD

### ISSUE THREAD

**Issue by qcolombet** [openxla/iree#11970]
### What happened?

I don't know if non-lld linkers are supposed to be supported on linux, but I forgot to set `-DIREE_ENABLE_LLD=ON` on a fresh build and was unable to build the tracy related tools (see the reproducer section for more details).

The issue is that the `libIREETracyNFD.a` library happens too early in the link list AFAICT.

The attached patch fixes that but I don't know if we care.
[tentative_fix.patch](https://github.com/iree-org/iree/files/10508140/tentative_fix.patch)

### Steps to reproduce your issue

To reproduce:
```
cmake -GNinja -DIREE_BUILD_TRACY=ON -DCMAKE_BUILD_TYPE=RelWithDebInfo  -DIREE_ENABLE_ASSERTIONS=OFF  <iree_checkout>
ninja iree-tracy-profiler iree-tracy-capture iree-tracy-csvexport
```

Result:
```
[1/1] Linking CXX executable tracy/iree-tracy-profiler
FAILED: tracy/iree-tracy-profiler 
: && /usr/bin/c++ -O2 -g -DNDEBUG  tracy/CMakeFiles/IREETracyProfiler.dir/__/__/__/third_party/tracy/profiler/src/BackendGlfw.cpp.o tracy/CMakeFiles/IREETracyProfiler.dir/__/__/__/third_party/tracy/profiler/src/ConnectionHistory.cpp.o tracy/CMakeFiles/IREETracyProfiler.dir/__/__/__/third_party/tracy/profiler/src/Filters.cpp.o tracy/CMakeFiles/IREETracyProfiler.dir/__/__/__/third_party/tracy/profiler/src/Fonts.cpp.o tracy/CMakeFiles/IREETracyProfiler.dir/__/__/__/third_party/tracy/profiler/src/HttpRequest.cpp.o tracy/CMakeFiles/IREETracyProfiler.dir/__/__/__/third_party/tracy/profiler/src/ImGuiContext.cpp.o tracy/CMakeFiles/IREETracyProfiler.dir/__/__/__/third_party/tracy/profiler/src/IsElevated.cpp.o tracy/CMakeFiles/IREETracyProfiler.dir/__/__/__/third_party/tracy/profiler/src/ResolvService.cpp.o tracy/CMakeFiles/IREETracyProfiler.dir/__/__/__/third_party/tracy/profiler/src/RunQueue.cpp.o tracy/CMakeFiles/IREETracyProfiler.dir/__/__/__/third_party/tracy/profiler/src/WindowPosition.cpp.o tracy/CMakeFiles/IREETracyProfiler.dir/__/__/__/third_party/tracy/profiler/src/main.cpp.o tracy/CMakeFiles/IREETracyProfiler.dir/__/__/__/third_party/tracy/profiler/src/winmain.cpp.o tracy/CMakeFiles/IREETracyProfiler.dir/__/__/__/third_party/tracy/profiler/src/winmainArchDiscovery.cpp.o -o tracy/iree-tracy-profiler  -ltbb  -lzstd  -lcapstone  -ldl  -lfreetype  -lglfw  -lgtk-3  -lgdk-3  -lpangocairo-1.0  -lpango-1.0  -lharfbuzz  -latk-1.0  -lcairo-gobject  -lcairo  -lgdk_pixbuf-2.0  -lgio-2.0  -lgobject-2.0  -lglib-2.0  -ldbus-1  tracy/libIREETracyIMGUI.a  tracy/libIREETracyCommon.a  tracy/libIREETracyNFD.a  tracy/libIREETracyServer.a  -lfreetype  -lglfw  -lgtk-3  -lgdk-3  -lpangocairo-1.0  -lpango-1.0  -lharfbuzz  -latk-1.0  -lcairo-gobject  -lcairo  -lgdk_pixbuf-2.0  -lgio-2.0  -lgobject-2.0  -lglib-2.0  -ldbus-1  tracy/libIREETracyCommon.a  -ltbb  -lzstd  -lcapstone  -ldl && :
/usr/bin/ld: tracy/libIREETracyServer.a(TracyFileselector.cpp.o): in function `tracy::Fileselector::OpenFile(char const*, char const*, std::function<void (char const*)>)':
<iree_checkout>/third_party/tracy/server/TracyFileselector.cpp:80: undefined reference to `NFD_OpenDialogN'
/usr/bin/ld: tracy/libIREETracyServer.a(TracyFileselector.cpp.o): in function `OpenFileImpl':
<iree_checkout>/third_party/tracy/server/TracyFileselector.cpp:84: undefined reference to `NFD_FreePathN'
/usr/bin/ld: tracy/libIREETracyServer.a(TracyFileselector.cpp.o): in function `tracy::Fileselector::SaveFile(char const*, char const*, std::function<void (char const*)>)':
<iree_checkout>/third_party/tracy/server/TracyFileselector.cpp:101: undefined reference to `NFD_SaveDialogN'
/usr/bin/ld: tracy/libIREETracyServer.a(TracyFileselector.cpp.o): in function `SaveFileImpl':
<iree_checkout>/third_party/tracy/server/TracyFileselector.cpp:105: undefined reference to `NFD_FreePathN'
/usr/bin/ld: tracy/libIREETracyServer.a(TracyFileselector.cpp.o): in function `tracy::Fileselector::Init()':
<iree_checkout>/third_party/tracy/server/TracyFileselector.cpp:19: undefined reference to `NFD_Init'
/usr/bin/ld: tracy/libIREETracyServer.a(TracyFileselector.cpp.o): in function `tracy::Fileselector::Shutdown()':
<iree_checkout>/third_party/tracy/server/TracyFileselector.cpp:26: undefined reference to `NFD_Quit'
collect2: error: ld returned 1 exit status
ninja: build stopped: subcommand failed
```

### What component(s) does this issue relate to?

Other

**Comment by ScottTodd** [comment]
To clarify the issue title a bit: `-DIREE_BUILD_TRACY=ON` is just a convenience utility that IREE provides to build Tracy's standalone tools without needing to switch to a different build system. YMMV if you stray too far off the path that other IREE developers are using to build that convenience target. "Tracing enabled" typically refers to `-DIREE_ENABLE_RUNTIME_TRACING=ON`, which includes Tracy instrumentation in IREE's runtime - and _that_ is something that we more officially support across build configurations.

The supported way to build Tracy is documented in Tracy's manual. Section 2.3.1.2 for Unix has these instructions:
> On Unix go to the build/unix directory and build the release target using GNU make.
(https://github.com/wolfpld/tracy/tree/master/profiler/build/unix)

**Comment by qcolombet** [comment]
> To clarify the issue title a bit: `-DIREE_BUILD_TRACY=ON` is just a convenience utility that IREE provides to build Tracy's standalone tools without needing to switch to a different build system. 

Sure, but I literally just followed our documentation here and it didn’t work:
https://github.com/iree-org/iree/blob/main/docs/developers/developing_iree/profiling_with_tracy.md#build-the-tracy-tools

**Comment by GMNGeoffrey** [comment]
Seems like this mostly a documentation issue? The developer docs are in many cases very out of date, unfortunately. @qcolombet could you please update them as you think they need to be updated? I honestly think it's a bad idea that we have a convenience target for building Tracy at all.

**Comment by allieculp** [comment]
@qcolombet Bumping this issue back up, are you working on the documentation? I think we can set this as an in progress P2 depending on your current bandwidth.

**Comment by qcolombet** [comment]
Thanks for the reminder.
Tentative fix at #13041
